### PR TITLE
Add config option for http endpoint

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

File: metricbeat/metricbeat.reference.yml [hunk 1]
```diff
@@ -1267,3 +1267,19 @@ logging.files:
 
 # Set to true to log messages in json format.
 #logging.json: false
+
+
+#================================ HTTP Endpoint ======================================
+# Each beat can expose internal data points through a http endpoint. For security
```
Comment (andrewkroh) [inline metricbeat/metricbeat.reference.yml]: - "data points" - "metrics" instead?
- Capitalize `http`.
- `s/reason/reasons/`
- Put a period after `/stats` to end the sentence.
- "For pretty JSON output append a `?pretty` query string to the URL.

File: metricbeat/metricbeat.reference.yml [hunk 2]
```diff
@@ -1267,3 +1267,19 @@ logging.files:
...
+# reason the endpoint is disabled by default. This feature is currently experimental.
+# Stats can be access through http://localhost:5066/stats For a pretty json output,
+# add ?pretty .
+
+# Defines if http endpoint is enabled
```
Comment (andrewkroh) [inline metricbeat/metricbeat.reference.yml]: `Defines if the HTTP endpoint is enabled.`

File: metricbeat/metricbeat.reference.yml [hunk 3]
```diff
@@ -1267,3 +1267,19 @@ logging.files:
...
+#http.enabled: false
+
+# Host to expose the http endpoint to. It is recommended to use only localhost.
```
Comment (andrewkroh) [inline metricbeat/metricbeat.reference.yml]: The HTTP endpoint will bind to this hostname or IP address.

File: metricbeat/metricbeat.reference.yml [hunk 4]
```diff
@@ -1267,3 +1267,19 @@ logging.files:
...
+#http.host: localhost
+
+# Port on which the http endpoint is exposed. Default is 5066
```
Comment (andrewkroh) [inline metricbeat/metricbeat.reference.yml]: Port on which the HTTP endpoint will bind. Default is 5066.

File: libbeat/_meta/config.reference.yml [hunk 5]
```diff
@@ -778,3 +778,19 @@ logging.files:
...
+
+
+#================================ HTTP Endpoint ======================================
+# Each beat can expose internal data points through a http endpoint. For security
+# reason the endpoint is disabled by default. This feature is currently experimental.
+# Stats can be access through http://localhost:5066/stats For a pretty json output,
+# add ?pretty .
+
+# Defines if http endpoint is enabled
+#http.enabled: false
```
Comment (exekias) [inline libbeat/_meta/config.reference.yml]: Would it make sense to put a more specific name here? `http` is pretty generic, what about `stats.http` or similar?

File: metricbeat/metricbeat.reference.yml (hunk 1 above)
Comment (ruflin) [reply metricbeat/metricbeat.reference.yml]: fixed

File: metricbeat/metricbeat.reference.yml (hunk 2 above)
Comment (ruflin) [reply metricbeat/metricbeat.reference.yml]: done

File: metricbeat/metricbeat.reference.yml (hunk 3 above)
Comment (ruflin) [reply metricbeat/metricbeat.reference.yml]: done

**Pull request merged**